libc/test/src/math/ddivl_test.cpp

```diff
@@ -0,0 +1,13 @@
+//===-- Unittests for ddivl -----------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "DivTest.h"
+
+#include "src/math/ddivl.h"
+
+LIST_ADD_TESTS(double, long double, LIBC_NAMESPACE::ddivl)
```

Comment:
fixed + it passes for me.

Reply:
Are you on an x86-64 or AArch64 machine?

Reply:
X86-64

Reply:
Interesting. I'll wait for the CI results.

Reply:
Build broke. Will check it out.

Reply:
Any thoughts as to why? I just changed the AddTest to DivTest for Hdrs in the tests cmakelists

Reply:
When math tests fail only for `long double`, it's usually because the format of `long double` on x86 needs special care. I would suggest you remove `ddivl` from this PR so that we can merge it. Then, if you want to fix `ddivl` yourself, you can read https://en.wikipedia.org/wiki/Extended_precision#x86_extended_precision_format.

I'm not sure why the tests passed on your machine. Could you enter these commands in a terminal and show the output?

```sh
builddir=replace/with/path/to/your/build/directory
cmake --build "${builddir}" --target libc.test.src.math.ddivl_test.__unit__.__build__
file "${builddir}"/projects/libc/test/src/math/libc.test.src.math.ddivl_test.__unit__.__build__
```

Reply:
@aaryanshukla Could you please run these commands and post the output?

Reply:
ninja: error: unknown target 'libc.test.src.math.ddivl_test.__unit__.__build__', did you mean 'libc.test.src.math.fdiml_test.__unit__.__build__'?
build/projects/libc/test/src/math/libc.test.src.math.ddivl_test.__unit__.__build__: cannot open `build/projects/libc/test/src/math/libc.test.src.math.ddivl_test.__unit__.__build__' (No such file or directory)

Reply:
I deleted and created build again, still same issue.

Reply:
When the CI failed on another PR before you added explicit template instantiations in MPFRUtils.cpp, did you not get that error while testing locally? Maybe the MPFR tests are just disabled in your build. They're disabled in full build mode.

Reply:
When turning full build off, for NO_FMA_OPT DDIVL I am getting all cases passed.

Reply:
file build/projects/libc/test/src/math/libc.test.src.math.ddivl_test.__unit__.__NO_FMA_OPT.__build__
build/projects/libc/test/src/math/libc.test.src.math.ddivl_test.__unit__.__NO_FMA_OPT.__build__: cannot open `build/projects/libc/test/src/math/libc.test.src.math.ddivl_test.__unit__.__NO_FMA_OPT.__build__' (No such file or directory)

Reply:
This is my ninja check-libc tests: Ran 5 tests.  PASS: 5  FAIL: 0
[795/1116] Running unit test libc.test.src.math.smoke.ddivl.__unit__.__NO_FMA_OPT
[==========] Running 5 tests from 1 test suite.
[ RUN      ] LlvmLibcDivTest.SpecialNumbers
[       OK ] LlvmLibcDivTest.SpecialNumbers (5 us)
[ RUN      ] LlvmLibcDivTest.DivisionByZero
[       OK ] LlvmLibcDivTest.DivisionByZero (3 us)
[ RUN      ] LlvmLibcDivTest.InvalidOperations
[       OK ] LlvmLibcDivTest.InvalidOperations (5 us)
[ RUN      ] LlvmLibcDivTest.RangeErrors
[       OK ] LlvmLibcDivTest.RangeErrors (12 us)
[ RUN      ] LlvmLibcDivTest.InexactResults
[       OK ] LlvmLibcDivTest.InexactResults (1 us)

Reply:
You should see a warning message saying "Math tests using MPFR will be skipped" in the CMake configuration output by the way. You can apply this patch and check the CMake configuration output to try to debug the issue:

```diff
diff --git a/libc/cmake/modules/LLVMLibCCheckMPFR.cmake b/libc/cmake/modules/LLVMLibCCheckMPFR.cmake
index a27c2dc0c030..3672d25d1378 100644
--- a/libc/cmake/modules/LLVMLibCCheckMPFR.cmake
+++ b/libc/cmake/modules/LLVMLibCCheckMPFR.cmake
@@ -14,5 +14,7 @@ else()
     ${LIBC_SOURCE_DIR}/utils/MPFRWrapper/check_mpfr.cpp
     LINK_LIBRARIES
       -lmpfr -lgmp -latomic
+    OUTPUT_VARIABLE foo
   )
+  message(WARNING ${foo})
 endif()
```

Reply:
Yeah I see the math tests using MPFR will be skipped, I need to install MPFR. Should I still merge this patch then try to solve the MPFR issues.

Reply:
The CI build passed after you removed `ddivl` so you can merge this PR.

Reply:
Is there documentation for installing mpfr.

Reply:
I'm not aware of any. If you're on Linux then you should search your Linux distribution's package repositories.